examples/summarize_scoring_using_bedrock.ipynb

```diff
@@ -0,0 +1,374 @@
+    "\n",
+    "The below sets up TestSuites + Runs for each unique combination so we have the ability to rank the models in a round-robin tournament for ELO rating. \n",
+    "\n",
+    "See the [Quickstart Guide](https://bench.readthedocs.io/en/latest/quickstart.html#view-results-in-local-ui) for additional information."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "89e53792-7c88-41d4-a778-29aeab195f3f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from arthur_bench.run.testsuite import TestSuite\n",
+    "from itertools import combinations\n",
+    "from collections import defaultdict\n",
+    "\n",
+    "\n",
+    "# Make sure you've set the BENCH_FILE_DIR environment variable\n",
+    "\n",
+    "# The summary_quality scorer uses ChatGPT to score the summary\n",
+    "# so make sure that your OPENAI_API_KEY environment variable is set\n",
+    "\n",
+    "combos = list(combinations(models, 2))\n",
+    "d_combos = defaultdict(list)\n",
+    "for m1, m2 in combos:\n",
+    "    d_combos[m1].append(m2)\n",
+    "\n",
+    "print(d_combos)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ec69abd6-5aa6-4a40-9f79-00f9505fcb59",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_test_suite(model):\n",
+    "    print(f\"Creating test suite for model {model}\")\n",
+    "    return TestSuite(\n",
+    "        f\"News Summarization using {model} as reference\", \n",
+    "        'summary_quality',\n",
+    "        input_text_list=summaries[\"input\"],\n",
+    "        reference_output_list=summaries[model]\n",
+    "    )    \n",
+    "\n",
+    "suites = {}\n",
+    "for model in d_combos.keys():\n",
+    "    suites[model] = create_test_suite(model)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "09afb482-cbb7-43fe-8f3b-804bd0691d2f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def run_test_suite(suite, ref_model, cand_model):\n",
+    "    run = suite.run(\n",
+    "        run_name=f\"{ref_model}_vs_{cand_model}\",\n",
+    "        candidate_output_list=summaries[cand_model]\n",
+    "    )\n",
+    "    return run\n",
+    "\n",
+    "runs = defaultdict(dict)\n",
+    "for ref_model, suite in suites.items():\n",
+    "    for cand_model in d_combos[ref_model]:\n",
+    "        runs[ref_model][cand_model] = run_test_suite(suite, ref_model, cand_model)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "id": "1f1185e4-b39d-4107-bbc4-3aedc449cf46",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[('anthropic.claude-v2', 1345.4153155340496), ('meta.llama2-70b-chat-v1', 1079.6979024278733), ('amazon.titan-text-lite-v1', 898.4211374051577), ('anthropic.claude-v1', 864.5103513782595), ('meta.llama2-13b-chat-v1', 811.9552932546615)]\n"
+     ]
+    }
+   ],
+   "source": [
```

Comment:
Yeah - it's pretty exciting. As you called out, we have to create a Test Suite to define the left-hand side of the comparison operation (where each test run defines the right-hand of the comparison). So in order for us to do a head-to-head comparison of each unique combination of models, we have to define a Test Suite per model (and some models will have more test runs than others). 

I also modeled each unique test-case (eg: single inference) as a "game" for the ELO scorer, but we could also consider the entire Run as a single game as well (eg: which model, aggregated across all inferences, performed better). I think the single inference case is better here because it's apples-to-apples (all models ran on the same set of evaluation data), but can see cases where it might be better to use the aggregated run for scoring. 

I think it'd be great for this ELO comparison to live as a first-class workflow within Bench (at the very least, a library to do this - even cooler would be a UI around it), but for now I think the notebook approach demonstrates it's at least doable and easy to set up. 

For this comment, is there anything you wanted me to do/change? Or was this more just calling out you generally are in favor of the approach?

Reply:
just calling out